arbitrator/stylus/src/target_cache.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022-2024, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE
+
+use eyre::{eyre, OptionExt, Result};
+use lazy_static::lazy_static;
+use parking_lot::RwLock;
+use std::{collections::HashMap, str::FromStr};
+use wasmer_types::{CpuFeature, Target, Triple};
+
+use crate::cache::InitCache;
+
+lazy_static! {
+    static ref TARGET_CACHE: RwLock<HashMap<String, Target>> = RwLock::new(HashMap::new());
+}
+
+fn target_from_string(input: String) -> Result<Target> {
+    let mut parts = input.split('+');
+
+    let Some(trip_sting) = parts.next() else {
+        return Err(eyre!("no architecture"));
+    };
+
+    let trip = match Triple::from_str(trip_sting) {
+        Ok(val) => val,
+        Err(e) => return Err(eyre!(e)),
+    };
+
+    let mut features = CpuFeature::set();
+    for flag in parts {
+        features.insert(CpuFeature::from_str(flag)?);
```

Comment:
what happens when `CpuFeature::from_str` fails? Are we ok ignoring the error here?

Reply:
When a ? encounters an error it doesn't ignore it, but immediately returns this error as a result of the function it's inside (so target_from_string will return the error)

Reply:
nice, missed that, thanks!